test/cpp/api/sequential.cpp

```diff
@@ -32,20 +32,47 @@ TEST_F(SequentialTest, ConstructsFromSharedPointer) {
   Sequential sequential(
       std::make_shared<M>(1), std::make_shared<M>(2), std::make_shared<M>(3));
   ASSERT_EQ(sequential->size(), 3);
+
+  Sequential sequential_named(modules_ordered_dict({
+    {"m1", std::make_shared<M>(1)},
+    {std::string("m2"), std::make_shared<M>(2)},
+    {"m3", std::make_shared<M>(3)}
+  }));
+  ASSERT_EQ(sequential->size(), 3);
 }
 
 TEST_F(SequentialTest, ConstructsFromConcreteType) {
+  static int copy_count;
```

Comment:
Ideally we want the `copy_count` to be inside the struct as a static member (so that it can keep track of the number of copy constructor calls happened for this class), however static data members are not allowed in local classes (https://stackoverflow.com/questions/8162994/why-arent-static-data-members-allowed-in-local-classes), and we have to declare it as a global static variable instead.

Reply:
is it global static or static local variable?  Also, that answered why it isn't static in the struct but why is it static at all?

Reply:
`copy_count` is within the scope of `SequentialTest::ConstructsFromConcreteType`, and won't be polluting other tests. If we define it as a non-static variable, it will throw `error: use of local variable with automatic storage from containing function` error (https://stackoverflow.com/questions/43932288/declared-class-inside-the-main-function-in-c), and the workaround is to either define it as a static variable (which is the chosen approach here) or pass it into the copy constructor.

Reply:
ah I see, makes sense.